### PR TITLE
1 is not composite

### LINKED ISSUE #8081: inconsistency between assumptions and isprime
Assumptions applies "prime" to only integers that are positive; that's the same as isprime...but if an integer is not prime and other than 0, 1, 2 then it is composite(?). But assumptions.py defines composite only for positive integers that are not prime. This should probably be mentioned in documentation of assumptions and/or isprime.

### LINKED ISSUE #8807: factorial(0).is_composite returns True, the answer however should be False
>>> factorial(0).is_composite
True

### PR DESCRIPTION
**Pull request by smichr**
replaces and closes #8809
closes #8807
closes #8081

### CODE REVIEW

**Comment by smichr** [comment]
In order to allow S(1).is_composite to be False *and* keep positive, integer, non-prime numbers composite, I've defined an `_eval_is_composite` routine for Expr. I'm not sure if this is ok to do or not. Alternatively, we could define as `is_one` attribute and then that could have `composite=False`. @asmeurer could probably comment on the merits.

**Comment by pelegm** [comment]
+1

**Comment by smichr** [comment]
Thanks, @pelegm , I'll commit this.

**Comment by asmeurer** [comment]
I wouldn't create is_one. We should focus on getting an assumptions system that can handle any kind of inequality. For example, another useful fact is that primes > 2 are odd, but we don't want to add is_two.

File: sympy/core/tests/test_assumptions.py [hunk 1]
```diff
@@ -481,6 +477,9 @@ def test_composite():
     assert S(2).is_composite is False
     assert S(17).is_composite is False
     assert S(4).is_composite is True
+    x = Dummy(integer=True, positive=True, prime=False)
+    assert x.is_composite
```
Comment (skirpichev) [inline sympy/core/tests/test_assumptions.py]: @smichr, as I said in the referenced pr, this is wrong!  Counterexample is x=S.One.

While it's ok to do bad reviews (all people do mistakes), you shouldn't silently ignore already mentioned flaws.  @asmeurer, @certik, I'm surprised that you don't care about this...

Comment (pelegm) [inline sympy/core/tests/test_assumptions.py]: This indeed looks wrong.

Comment (smichr) [inline sympy/core/tests/test_assumptions.py]: > you shouldn't silently ignore

Sorry to have missed this. After opening a PR I look for feedback there. I don't remember all the reasons that this was committed then, but Aaron (who also commented on the PR but after the commit -- I didn't see it) has made the correction and I have committed it.

**Pull request merged**